### ISSUE TITLE
[BUG] Different behaviour between a normal compilation and a compilation with the `--profile=release`

### ISSUE THREAD

**Issue by xvw** [ocsigen/js_of_ocaml#1522]
**Describe the bug**
I notice a different behaviour between a normal compilation and a compilation with the `--profile=release` flag for handling rejections in promises. I haven't really managed to isolate the problem so I'm sorry for the limited information I can provide. This is always the case when developing an application that interacts with the [Tezos](https://tezos.com) chain. Here's the difference between "normal" compilation (in development) and compilation with the `--profile=release` flag:

### compiling without `--profile=release`

Here the application is launched in dev mode, the bundle is ~16mb
![image](https://github.com/ocsigen/js_of_ocaml/assets/5732466/dd6b9f65-79a1-4b8e-b8e6-42282f7ba723)

I can click on syncrhonize Wallet
![image](https://github.com/ocsigen/js_of_ocaml/assets/5732466/51148e1d-7a68-4708-b407-d14d3521de9b)

After some step connecting through the wallet, everything is fine:
![image](https://github.com/ocsigen/js_of_ocaml/assets/5732466/fd537d5a-c2d7-4b39-a3c1-65085071c5b8)

(And the application works)

### compiling with `--profile=release`

Here the application is launched in `release` mode, the bundle is ~1.20mb
![image](https://github.com/ocsigen/js_of_ocaml/assets/5732466/ec74f22c-91a0-48d3-83fc-5f3f6a5446f1)

I have the same first screen (which is an huge progress since the new patches)
![image](https://github.com/ocsigen/js_of_ocaml/assets/5732466/893b1075-fac9-4341-90b8-240c8f9914e8)

And I follow the instruction like in "dev" mode, but when I select my wallet, I face to : 
![image](https://github.com/ocsigen/js_of_ocaml/assets/5732466/a1a3ec4d-5778-4d7e-96a4-d9ec91da0896)

So the error is a Promise rejection, which does not appear in "dev" mode. Here is the stack:

```
18:58:26.784 request-permissions-rejection, Nightmare_js__Promise.Promise_rejection(_) capsule_client.bc.js:9868:329
    dB http://localhost:8888/js/capsule_client.bc.js:9868
    h http://localhost:8888/js/capsule_client.bc.js:7715
    c_ http://localhost:8888/js/capsule_client.bc.js:9868
    n http://localhost:8888/js/capsule_client.bc.js:12493
    h http://localhost:8888/js/capsule_client.bc.js:7715
    f http://localhost:8888/js/capsule_client.bc.js:9735
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    po http://localhost:8888/js/capsule_client.bc.js:9717
    f http://localhost:8888/js/capsule_client.bc.js:9724
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    po http://localhost:8888/js/capsule_client.bc.js:9717
    f http://localhost:8888/js/capsule_client.bc.js:9724
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    po http://localhost:8888/js/capsule_client.bc.js:9717
    g http://localhost:8888/js/capsule_client.bc.js:9755
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    f http://localhost:8888/js/capsule_client.bc.js:9733
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    pk http://localhost:8888/js/capsule_client.bc.js:9705
    d7 http://localhost:8888/js/capsule_client.bc.js:9847
    e http://localhost:8888/js/capsule_client.bc.js:7362
    (Async: promise callback)
    aFi http://localhost:8888/js/capsule_client.bc.js:7363
    d7 http://localhost:8888/js/capsule_client.bc.js:9847
    m http://localhost:8888/js/capsule_client.bc.js:12499
    h http://localhost:8888/js/capsule_client.bc.js:7715
    kl http://localhost:8888/js/capsule_client.bc.js:9747
    e http://localhost:8888/js/capsule_client.bc.js:12499
    h http://localhost:8888/js/capsule_client.bc.js:7715
    f http://localhost:8888/js/capsule_client.bc.js:9722
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    f http://localhost:8888/js/capsule_client.bc.js:9737
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    pk http://localhost:8888/js/capsule_client.bc.js:9705
    d http://localhost:8888/js/capsule_client.bc.js:9847
    f http://localhost:8888/js/capsule_client.bc.js:7361
    (Async: promise callback)
    aFi http://localhost:8888/js/capsule_client.bc.js:7363
    d7 http://localhost:8888/js/capsule_client.bc.js:9847
    J http://localhost:8888/js/capsule_client.bc.js:12502
    h http://localhost:8888/js/capsule_client.bc.js:7715
    hK http://localhost:8888/js/capsule_client.bc.js:9768
    J http://localhost:8888/js/capsule_client.bc.js:12483
    o http://localhost:8888/js/capsule_client.bc.js:7716
    lx http://localhost:8888/js/capsule_client.bc.js:12025
    lx http://localhost:8888/js/capsule_client.bc.js:12020
    h http://localhost:8888/js/capsule_client.bc.js:7715
    V http://localhost:8888/js/capsule_client.bc.js:7901
    lx http://localhost:8888/js/capsule_client.bc.js:12020
    D http://localhost:8888/js/capsule_client.bc.js:12465
    m http://localhost:8888/js/capsule_client.bc.js:12462
    h http://localhost:8888/js/capsule_client.bc.js:7715
    m http://localhost:8888/js/capsule_client.bc.js:12440
    o http://localhost:8888/js/capsule_client.bc.js:7716
    cG http://localhost:8888/js/capsule_client.bc.js:7905
    e http://localhost:8888/js/capsule_client.bc.js:12439
    h http://localhost:8888/js/capsule_client.bc.js:7715
    jC http://localhost:8888/js/capsule_client.bc.js:7867
    e http://localhost:8888/js/capsule_client.bc.js:12426
    dp http://localhost:8888/js/capsule_client.bc.js:6648
    m6 http://localhost:8888/js/capsule_client.bc.js:7160
    (Async: EventListener.handleEvent)
    H http://localhost:8888/js/capsule_client.bc.js:12456
    h http://localhost:8888/js/capsule_client.bc.js:7715
    sT http://localhost:8888/js/capsule_client.bc.js:12068
    h http://localhost:8888/js/capsule_client.bc.js:7715
    V http://localhost:8888/js/capsule_client.bc.js:7901
    sT http://localhost:8888/js/capsule_client.bc.js:12060

18:58:26.784 Impossible de connecter le wallet capsule_client.bc.js:9868:329
    dB http://localhost:8888/js/capsule_client.bc.js:9868
    h http://localhost:8888/js/capsule_client.bc.js:7715
    c_ http://localhost:8888/js/capsule_client.bc.js:9868
    n http://localhost:8888/js/capsule_client.bc.js:12493
    h http://localhost:8888/js/capsule_client.bc.js:7715
    f http://localhost:8888/js/capsule_client.bc.js:9735
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    po http://localhost:8888/js/capsule_client.bc.js:9717
    f http://localhost:8888/js/capsule_client.bc.js:9724
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    po http://localhost:8888/js/capsule_client.bc.js:9717
    f http://localhost:8888/js/capsule_client.bc.js:9724
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    po http://localhost:8888/js/capsule_client.bc.js:9717
    g http://localhost:8888/js/capsule_client.bc.js:9755
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    f http://localhost:8888/js/capsule_client.bc.js:9733
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    pk http://localhost:8888/js/capsule_client.bc.js:9705
    d7 http://localhost:8888/js/capsule_client.bc.js:9847
    e http://localhost:8888/js/capsule_client.bc.js:7362
    (Async: promise callback)
    aFi http://localhost:8888/js/capsule_client.bc.js:7363
    d7 http://localhost:8888/js/capsule_client.bc.js:9847
    m http://localhost:8888/js/capsule_client.bc.js:12499
    h http://localhost:8888/js/capsule_client.bc.js:7715
    kl http://localhost:8888/js/capsule_client.bc.js:9747
    e http://localhost:8888/js/capsule_client.bc.js:12499
    h http://localhost:8888/js/capsule_client.bc.js:7715
    f http://localhost:8888/js/capsule_client.bc.js:9722
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    f http://localhost:8888/js/capsule_client.bc.js:9737
    h http://localhost:8888/js/capsule_client.bc.js:7715
    d http://localhost:8888/js/capsule_client.bc.js:9690
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    pg http://localhost:8888/js/capsule_client.bc.js:9693
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    h http://localhost:8888/js/capsule_client.bc.js:7715
    ph http://localhost:8888/js/capsule_client.bc.js:9694
    dA http://localhost:8888/js/capsule_client.bc.js:9697
    pk http://localhost:8888/js/capsule_client.bc.js:9705
    d http://localhost:8888/js/capsule_client.bc.js:9847
    f http://localhost:8888/js/capsule_client.bc.js:7361
    (Async: promise callback)
    aFi http://localhost:8888/js/capsule_client.bc.js:7363
    d7 http://localhost:8888/js/capsule_client.bc.js:9847
    J http://localhost:8888/js/capsule_client.bc.js:12502
    h http://localhost:8888/js/capsule_client.bc.js:7715
    hK http://localhost:8888/js/capsule_client.bc.js:9768
    J http://localhost:8888/js/capsule_client.bc.js:12483
    o http://localhost:8888/js/capsule_client.bc.js:7716
    lx http://localhost:8888/js/capsule_client.bc.js:12025
    lx http://localhost:8888/js/capsule_client.bc.js:12020
    h http://localhost:8888/js/capsule_client.bc.js:7715
    V http://localhost:8888/js/capsule_client.bc.js:7901
    lx http://localhost:8888/js/capsule_client.bc.js:12020
    D http://localhost:8888/js/capsule_client.bc.js:12465
    m http://localhost:8888/js/capsule_client.bc.js:12462
    h http://localhost:8888/js/capsule_client.bc.js:7715
    m http://localhost:8888/js/capsule_client.bc.js:12440
    o http://localhost:8888/js/capsule_client.bc.js:7716
    cG http://localhost:8888/js/capsule_client.bc.js:7905
    e http://localhost:8888/js/capsule_client.bc.js:12439
    h http://localhost:8888/js/capsule_client.bc.js:7715
    jC http://localhost:8888/js/capsule_client.bc.js:7867
    e http://localhost:8888/js/capsule_client.bc.js:12426
    dp http://localhost:8888/js/capsule_client.bc.js:6648
    m6 http://localhost:8888/js/capsule_client.bc.js:7160
    (Async: EventListener.handleEvent)
    H http://localhost:8888/js/capsule_client.bc.js:12456
    h http://localhost:8888/js/capsule_client.bc.js:7715
    sT http://localhost:8888/js/capsule_client.bc.js:12068
    h http://localhost:8888/js/capsule_client.bc.js:7715
    V http://localhost:8888/js/capsule_client.bc.js:7901
    sT http://localhost:8888/js/capsule_client.bc.js:12060

```

Here is ... the line (I guess)

```js
o=function(a){return aD.detachEvent(f,g)}}j[1]=o;return c8(l,x)},aiJ=function(a,b){a.nightmare.suspending.push(b);return 0},hN=ac4(0),aiK={nightmare:{suspending:jo([0])},suspend:jq(aiJ),mount:jq(aiI)},kt=function(a){return hN.log(a)},aiM=function(a){return hN.info(a)},pB=function(a){return hN.warn(a)},dB=function(a){return hN.error(a)},c_=function(a,b){return h(a,L(b))},aiW=function(a){var
```

And as always, here is the full javascript bundle : https://gist.github.com/xvw/2a7c83da66b153972f33b0feedfac339

It's difficult to give more context, so here are the packages used:
- here are the different library for the VDom part https://github.com/funkywork/nightmare (with our Promise implementation here: https://github.com/funkywork/nightmare/blob/main/lib/js/promise.ml that relay on that javascript stubs: https://github.com/funkywork/nightmare/blob/main/lib/js/stubs/caml_promise_stubs.js)
- here are the different library for de Tezos Part https://github.com/funkywork/yourbones (with the specific binding of Beacon, the wallet abstraction that lives here https://github.com/funkywork/yourbones/tree/main/lib/beacon)

A side, here is the "dev mode" generated bundle (if it helps) https://gist.github.com/xvw/f814646dce98d3a85f5e1880a1631517

Thanks a lot, if I can give more context, feel free to ask.

**Versions**
I still use the master version of JSOO, everything is defined here: https://github.com/xvw/capsule/blob/main/capsule.opam

**Comment by xvw** [comment]
More context, using the flag `--pretty` (that gives this file https://gist.github.com/xvw/8a90470ca68c9876b61ffbdd34717f8b) the error is gone. So it seems that it is an error from the minimizer. 
Using `--disable inline` (without the flag `pretty`) I still have the error `11032:70` (using that file : https://gist.github.com/xvw/e2b7ea47b4743babb3d68facf9087b7a

**Comment by hhugo** [comment]
I spotted a bug in the variable renaming logic regarding class. fixed in #1524. I've no idea if this is related to your issue.

**Comment by xvw** [comment]
as mentionned with @hhugo, disabling `shortvar` fixe the issue (generated bundle https://gist.github.com/xvw/97683af0d6f866224a8833b743848830) !
